apps/frontend/src/libs/components/input/input.tsx

```diff
@@ -5,35 +5,76 @@ import {
 	type FieldValues,
 } from "react-hook-form";
 
-import { useFormController } from "~/libs/hooks/hooks.js";
+import { useCallback, useFormController } from "~/libs/hooks/hooks.js";
 
 type Properties<T extends FieldValues> = {
+	className: string;
+	clickIcon?: (event_: React.BaseSyntheticEvent) => void;
 	control: Control<T, null>;
 	errors: FieldErrors<T>;
+	hasIcon?: string;
+	iconSrc?: string;
+	inputChange?: (event_: React.BaseSyntheticEvent) => void;
 	label: string;
 	name: FieldPath<T>;
 	placeholder?: string;
-	type?: "email" | "text";
+	required?: string;
```

Comment:
why not just pass required as a boolean?

Reply:
yes you're right